.github/workflows/update-public-docs.yml

```diff
@@ -59,31 +62,85 @@ jobs:
           prerelease: false
 
       - name: Upload Release Asset
-        if: ${{ steps.check_docs.outputs.DOCS_CHANGED }}
-        id: upload-release-asset
+        if: ${{ steps.check_release_changed.outputs.DOCS_CHANGED }}
         uses: actions/upload-release-asset@v1
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         with:
-          upload_url: ${{ steps.create-release.outputs.upload_url }}
+          upload_url: ${{ steps.create_release.outputs.upload_url }}
           asset_path: ./new_public_docs.json
           asset_name: public_docs.json
           asset_content_type: application/json
 
-      - name: Trigger update in docs repo
-        if: ${{ steps.check_docs.outputs.DOCS_CHANGED }}
+      - name: Clone docs repo
+        uses: actions/checkout@v3
+        with:
+          repository: opensafely/documentation
+          path: docs
+          ref: 'main'
+          token: ${{ secrets.DOCS_WRITE_TOKEN }}
+
+      - name: Create or checkout branch
+        run: |
+          cd docs
+          git config --local user.email "action@github.com"
+          git config --local user.name "GitHub Action"
+          if [[ $(git ls-remote --heads origin $BRANCH) ]]; then
```

Comment:
nit: am interested in why you went with a subshell and output capturing here?

`if cmd` will branch on whether the `cmd` exits with a 0 or not. `git ls-remote` exits with a 1 if the branch doesn't exist, so the naked command should be all we need?

Not necessarily a problem, but if you want to check the output of `git ls-remote`, I think it's better spelt as `if test -n "$(git ls-remote ..)"`. Bash's [[ is weird and surprising (how it's implemented will horrify you), and I think "test -n" (which would be test --nonzerooutput if test did long options) is more explicit that [['s default.

But feel free to ignore this!

Reply:
The suggestion you made on the previous PR didn't work (it always ended up evaluating true), and my bash knowledge is only good enough to trial and error find a thing that works